masonry/src/widget/progress_bar.rs

```diff
@@ -0,0 +1,288 @@
+    }
+
+    fn on_text_event(&mut self, _ctx: &mut EventCtx, _event: &TextEvent) {}
+
+    fn on_access_event(&mut self, ctx: &mut EventCtx, event: &AccessEvent) {
+        if event.target == ctx.widget_id() {
+            match event.action {
+                // access events unhandled for now
+                _ => {}
+            }
+        }
+    }
+
+    fn on_status_change(&mut self, ctx: &mut LifeCycleCtx, _event: &StatusChange) {
+        ctx.request_paint();
+    }
+
+    fn lifecycle(&mut self, _ctx: &mut LifeCycleCtx, _event: &LifeCycle) {}
+
+    fn layout(&mut self, ctx: &mut LayoutCtx, bc: &BoxConstraints) -> Size {
+        const DEFAULT_WIDTH: f64 = 400.;
+
+        if self.label.needs_rebuild() {
+            let (font_ctx, layout_ctx) = ctx.text_contexts();
+            self.label.rebuild(font_ctx, layout_ctx);
+        }
+        let label_size = self.label.size();
+
+        let desired_size = Size::new(
+            DEFAULT_WIDTH.max(label_size.width),
+            crate::theme::BASIC_WIDGET_HEIGHT.max(label_size.height),
+        );
+        let our_size = bc.constrain(desired_size);
+        trace!("Computed layout: size={}", our_size);
+        our_size
+    }
+
+    fn paint(&mut self, ctx: &mut PaintCtx, scene: &mut Scene) {
+        let border_width = 1.;
+
+        if self.label.needs_rebuild() {
+            debug_panic!("Called ProgressBar paint before layout");
+        }
+
+        let rect = ctx
+            .size()
+            .to_rect()
+            .inset(-border_width / 2.)
+            .to_rounded_rect(2.);
+
+        fill_lin_gradient(
+            scene,
+            &rect,
+            [theme::BACKGROUND_LIGHT, theme::BACKGROUND_DARK],
+            UnitPoint::TOP,
+            UnitPoint::BOTTOM,
+        );
+
+        stroke(scene, &rect, theme::BORDER_DARK, border_width);
+
+        let progress_rect_size = Size::new(
+            ctx.size().width * self.progress.unwrap_or(1.) as f64,
+            ctx.size().height,
+        );
+        let progress_rect = progress_rect_size
+            .to_rect()
+            .inset(-border_width / 2.)
+            .to_rounded_rect(2.);
+
+        fill_lin_gradient(
+            scene,
+            &progress_rect,
+            [theme::PRIMARY_LIGHT, theme::PRIMARY_DARK],
+            UnitPoint::TOP,
+            UnitPoint::BOTTOM,
+        );
+        stroke(scene, &progress_rect, theme::BORDER_DARK, border_width);
+
+        // center text
+        let widget_size = ctx.size();
+        let label_size = self.label.size();
+        let text_pos = Point::new(
+            ((widget_size.width - label_size.width) * 0.5).max(0.),
+            ((widget_size.height - label_size.height) * 0.5).max(0.),
+        );
+        self.label.draw(scene, text_pos);
+    }
+
+    fn accessibility_role(&self) -> Role {
+        Role::ProgressIndicator
+    }
+
+    fn accessibility(&mut self, ctx: &mut AccessCtx) {
+        ctx.current_node().set_value(self.value_accessibility());
```

Comment:
Sorry, I made a typo last time. I meant [set_numeric_value](https://docs.rs/accesskit/latest/accesskit/struct.NodeBuilder.html#method.set_numeric_value).

```suggestion
        ctx.current_node().set_value(self.value_accessibility());
        if let Some(value) = self.progress {
            ctx.current_node().set_numeric_value(value * 100.0);
        }
```